datacube/ui/common.py

```diff
@@ -53,3 +74,109 @@ def find_any_metadata_suffix(path):
+
+    if depth < max_depth:
+        if show_sources:
+            info['sources'] = {key: build_dataset_info(index, source,
+                                                       show_sources=True, show_derived=False,
+                                                       depth=depth + 1, max_depth=max_depth)
+                               for key, source in dataset.sources.items()}
+
+        if show_derived:
+            info['derived'] = [build_dataset_info(index, derived,
+                                                  show_sources=False, show_derived=True,
+                                                  depth=depth + 1, max_depth=max_depth)
+                               for derived in index.datasets.get_derived(dataset.id)]
+
+    return info
+
+
+def _write_csv(infos):
+    writer = csv.DictWriter(sys.stdout, ['id', 'status', 'product', 'location'], extrasaction='ignore')
+    writer.writeheader()
+
+    def add_first_location(row):
+        locations_ = row['locations']
+        row['location'] = locations_[0] if locations_ else None
+        return row
+
+    writer.writerows(add_first_location(row) for row in infos)
+
+
+def _write_yaml(infos):
+    """
+    Dump yaml data with support for OrderedDicts.
+
+    Allows for better human-readability of output: such as dataset ID field first, sources last.
+
+    (Ordered dicts are output identically to normal yaml dicts: their order is purely for readability)
+    """
+
+    # We can't control how many ancestors this dumper API uses.
+    # pylint: disable=too-many-ancestors
+    class OrderedDumper(yaml.SafeDumper):
+        pass
+
+    def _dict_representer(dumper, data):
+        return dumper.represent_mapping(yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG, data.items())
+
+    def _range_representer(dumper, data):
+        # type: (yaml.Dumper, Range) -> Node
+        begin, end = data
+
+        # pyyaml doesn't output timestamps in flow style as timestamps(?)
+        if isinstance(begin, datetime.datetime):
+            begin = begin.isoformat()
+        if isinstance(end, datetime.datetime):
+            end = end.isoformat()
+
+        return dumper.represent_mapping(
+            yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG,
+            (('begin', begin), ('end', end)),
+            flow_style=True
+        )
+
+    def _reduced_accuracy_decimal_representer(dumper, data):
+        # type: (yaml.Dumper, Decimal) -> Node
+        return dumper.represent_float(
+            float(data)
+        )
+
+    OrderedDumper.add_representer(OrderedDict, _dict_representer)
+    OrderedDumper.add_representer(Range, _range_representer)
+    OrderedDumper.add_representer(Decimal, _reduced_accuracy_decimal_representer)
+    return yaml.dump_all(infos, sys.stdout, OrderedDumper, default_flow_style=False, indent=4)
+
+
+_OUTPUT_WRITERS = {
+    'csv': _write_csv,
+    'yaml': _write_yaml,
+}
+
+OUTPUT_FORMAT_OPT = click.option('-f', 'format_', help='Output format',
```

Comment:
I don't think this needs to be all caps. It's inconsistent with our other click decorators, and with typical python conventions which would indicate it's a constant.